File: src/commands/LastfmCommand.js
const Command = require('../structures/Command')
const fetch = require('node-fetch')
const { MessageAttachment } = require('discord.js')

class LastfmCommand extends Command {
  constructor () {
    super('lastfm')
    this.category = 'music'
  }

  async run (message, args, server, { t }) {
    if (!args[0]) return this.justify(message)
    const themes = {
      'duotone/purplish': ['duotone', 'purplish'],
      'duotone/natural': ['duotone', 'natural'],
      'duotone/divergent': ['duotone', 'divergent'],
      'duotone/bright-sun': ['duotone', 'sun'],
      'duotone/yellish': ['duotone', 'yellish'],
      'duotone/horror': ['duotone', 'horror']
    }
    const tops = {
      artists: ['MOST LISTENED ARTISTS', 'artists'],
      albums: ['MOST LISTENED ALBUMS', 'albums'],
      tracks: ['MOST LISTENED TRACKS', 'tracks']
    }
    const theme = themes[args[2]] ? themes[args[2]] : ['duotone', 'natural']
    const top = tops[args[1]] ? tops[args[1]] : ['MOST LISTENED TRACKS', 'tracks']
    fetch('https://generator.musicorumapp.com/generate', {
      method: 'post',
      headers: {
        'Content-Type': 'application/json',
        'X-Source': 'meteora'
      },
      body: JSON.stringify({ theme: theme[0], options: { user: args[0], period: '6month', top: top[1], pallete: theme[1], story: false, messages: { scrobbles: ['scrobbles', 'last 6 months'], subtitle: 'last 7 days', title: top[0] } } })
    }).then(async response => {
      const res = await response.json()
      const buffer = Buffer.from(res.base64.replace('data:image/jpeg;base64,', ''), 'base64')
      message.channel.send(new MessageAttachment(buffer))
    })
  }
}

module.exports = LastfmCommand

File: src/main/kotlin/com/mystery/meteora/client/commands/LastFMCommand.kt
package com.mystery.meteora.client.commands

import com.beust.klaxon.Klaxon
import com.beust.klaxon.KlaxonException
import com.mystery.meteora.controller.Config
import com.mystery.meteora.controller.Helper
import com.mystery.meteora.controller.lastFmApi.Response
import com.mystery.meteora.controller.translate
import com.mystery.meteora.handler.annotations.Command
import com.mystery.meteora.handler.annotations.Description
import com.mystery.meteora.handler.annotations.Module
import com.mystery.meteora.handler.annotations.Usage
import com.mystery.meteora.handler.modules.BaseModule
import net.dv8tion.jda.api.EmbedBuilder
import net.dv8tion.jda.api.events.message.MessageReceivedEvent
import okhttp3.MediaType
import okhttp3.OkHttpClient
import okhttp3.Request
import okhttp3.RequestBody
import java.awt.Color
import java.util.*

@Module("lastfm", "music")
@Usage("lastfm.usage")
@Description("lastfm.description")

class LastFMCommand(ctx: MessageReceivedEvent, args: String, prefix: String, config: Config) :
  BaseModule(ctx, args, prefix, config) {
  @Command("lastfm", "lfm")
  fun lastfm() {
    if (args == "") {
      Helper().explain(context, "lastfm", "lastfm", prefix, config!!)
      return
    }
    val client = OkHttpClient().newBuilder().build()
    val mediaType = MediaType.parse("application/json")
    val body = RequestBody.create(
      mediaType,
      "{\n\"theme\": \"duotone\",\n\"options\": " +
          "{\n\"user\": \"$args\",\n\"period\": \"7day\",\n\"top\": \"tracks\"," +
          "\n\"pallete\": \"sea\",\n\"story\": false,\n\"messages\": {\n\"scrobbles\": " +
          "[\n\"scrobbles\",\n\"last 7 days\"\n],\n\"subtitle\": \"last 7 days\"," +
          "\n\"title\": \"MOST LISTENED TRACKS\"\n}\n}\n}"
    )
    val request = Request.Builder()
      .url("https://generator.musicorumapp.com/generate")
      .method("POST", body)
      .addHeader("Content-Type", "application/json")
      .addHeader("X-Source", "meteora caralho")
      .build()
    val response = client.newCall(request).execute()
    val base64 = response.body()!!.string().replace("data:image/jpeg;base64,", "")
    try {
      val parsedResponse = Klaxon().parse<Response>(base64)!!
      val byteImage = Base64.getDecoder().decode(parsedResponse.base64)
      val embed = EmbedBuilder()
        .setImage("attachment://${context.author.id}.png")
        .setColor(Color(104, 235, 193))
      context.channel.sendFile(byteImage, "${context.author.id}.png").embed(embed.build()).queue()
    } catch (e: KlaxonException) {
      context.channel.sendMessage("lastfm.noUser".translate(config!!, context.guild.id, args)).queue()
    }
  }
}